### PR TITLE
3 - Import configs with texture

### LINKED ISSUE #15: Import a zip file (that we already exported)
Sometimes we need to share the particle that we created and exported to others,
Then they can import and work on it.
So we need to have a feature in Particular to allow us to import a zip file (which contains the JSON file and also the textures)

Or maybe we can have our own file type to write the JSON content + textures inside it (as a base64 or raw bytes)

### PR DESCRIPTION
**Pull request by manjav**

### CODE REVIEW

File: editor/lib/data/particular_editor_config.dart
Comment (tavangar79) [inline editor/lib/data/particular_editor_config.dart:1]: use spread operator instead ...addAll like

return {
...startColorMap,
}

File: editor/lib/data/particular_editor_controller.dart [hunk 1]
```diff
@@ -1,122 +1,55 @@
...
+    final configs = ParticularConfigs()..initialize(configs: configsData);
+
+    final layer = ParticularEditorLayer(
+      texture: texture ?? await getDefaultTexture(),
+      textureBytes: defaultTextureBytes,
+      configs: configs,
+    );
+
+    if (configsData == null || !configsData.containsKey("configName")) {
+      configs.updateFromMap({"configName": "Layer ${layers.length + 1}"});
```
Comment (tavangar79) [inline editor/lib/data/particular_editor_controller.dart]: can you change updateFromMap to copyWith and maybe with factory method

File: editor/lib/display/header_view.dart [hunk 2]
```diff
@@ -111,7 +114,26 @@ class _HeaderViewState extends State<HeaderView> {
...
+    for (var entry in files.entries) {
+      if (!entry.key.endsWith(".json")) {
+        continue;
+      }
+      List configLayers = entry.value;
+      for (var configLayer in configLayers) {
+        final layer = ParticularEditorLayer(
+          texture: files[configLayer["textureFileName"]].$1,
+          textureBytes: files[configLayer["textureFileName"]].$2,
+          configs: ParticularConfigs()..initialize(configs: configLayer),
```
Comment (tavangar79) [inline editor/lib/display/header_view.dart]: ParticularConfigs()..initialize(configs: configLayer) change to factory method for directly use ParticularConfigs.initialize

File: editor/lib/data/particular_editor_controller.dart (hunk 1 above)
Comment (manjav) [reply editor/lib/data/particular_editor_controller.dart]: This method is not copy or clone method. I change to `updateWith`. but if is there better example I really appreciated from.

File: editor/lib/display/header_view.dart (hunk 2 above)
Comment (manjav) [reply editor/lib/display/header_view.dart]: Done

File: editor/lib/data/particular_editor_config.dart
Comment (manjav) [reply editor/lib/data/particular_editor_config.dart:1]: Done

**Pull request merged**